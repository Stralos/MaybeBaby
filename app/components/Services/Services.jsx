// @flow
import React, { Component } from "react";
import { connect } from 'react-redux';
import { Link, withRouter } from "react-router-dom"
import type { Connector} from "react-redux";

type Props = {greeting: string, date: number};


class Services extends Component {
  props: Props

  render() {
    return(
      <div>
        <div> {this.props.greeting}</div>
        <div> {this.props.date}</div>
        <Link to={"/"}> Home </Link>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return Object.assign({}, {
  });
}

const mapSateToProps = (state) => {
  return Object.assign({}, {
    date: state.date,
    greeting: state.greeting
  })
};

const connector : Connector<{}, Props> = connect(mapSateToProps, mapDispatchToProps);

export default withRouter(connector(Services));
//export default withRouter(connect(mapSateToProps)(Services));

