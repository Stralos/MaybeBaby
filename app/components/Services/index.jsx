// @flow
import React, { Component } from "react";
import { connect } from 'react-redux';
import { Link, withRouter } from "react-router-dom"

type Props = { greeting: string, date: number };

class Services extends Component<void, Props, void> {
  render() {
    debugger;
    return(
      <div>
        <div> {this.props.greeting}</div>
        <div> {this.props.date}</div>
        <Link to={"/"}> Home </Link>
      </div>
    )
  }
}

const mapSateToProps = (state) : Props => {
  debugger;
  return Object.assign({}, {
    greeting: state.greeting,
    date: state.date
  })};

export default Services;
//export default withRouter(connect(mapSateToProps)(Services));

