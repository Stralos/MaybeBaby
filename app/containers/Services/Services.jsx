/* @flow */
import React, { PureComponent } from 'react';
import styled from 'styled-components';

import H3 from 'components/H3';
import Button from 'components/Button';

export type Product = {
  productName: string,
  range: boolean,
  minDurationMinutes: string,
  maxDurationMinutes?: string
}
export type Props = {
  services: Array<{
    name: string,
    products: Array<Product>
  }>
}

export const ProductList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

export const ProductDetails = styled.li`
  display: flex;
  justify-content: spaced-out;
  border-bottom: 1px solid #555;
  box-sizing: border-box;
  padding: 8px 0;
`;

export const ServicesList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

export const ServiceName = styled.h4`
  margin-top: 40px;
  margin-bottom: 24px;
  display: flex;
  justify-content: center;
  font-size: 1.4em;
  font-family: "Marcellus SC",serif;
`;

export const Service = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
`;

export const ProductName = styled.div`
  flex: 1;
  position: relative;
`;

export default class Services extends PureComponent<Props> {
  renderProducts = (products: Array<Product>) => {
    return products.map((product) => {
      const { productName } = product;
      return (
        <ProductDetails key={productName}>
          <ProductName>
            { productName }
          </ProductName>
          <Button>
            Book
          </Button>
        </ProductDetails>
      );
    });
  }

  renderServices = () => {
    const { services } = this.props;
    return services.map((service) => {
      const { name, products } = service;
      return (
        <Service key={name}>
          <ServiceName>
            { name }
          </ServiceName>
          <ProductList>
            { this.renderProducts(products) }
          </ProductList>
        </Service>
      );
    });
  }

  render = () => {
    return (
      <div>
        <H3>
          Services & Prices
        </H3>
        <ServicesList>
          { this.renderServices() }
        </ServicesList>
      </div>
    );
  }
}
