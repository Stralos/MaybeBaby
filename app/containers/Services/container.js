/* @flow */
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import type { Connector } from 'react-redux';

import Services from './Services';
import type { Props } from './Services';

const mapSateToProps = (state: State) => Object.assign({}, {
  services: state.services,
});

const connector : Connector<{}, Props> = connect(mapSateToProps, {});

export default withRouter(connector(Services));
