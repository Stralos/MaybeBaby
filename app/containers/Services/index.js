// @flow
import Services from './container';

export default Services;
