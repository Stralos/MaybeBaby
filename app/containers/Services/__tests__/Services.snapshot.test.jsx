/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Services from '../Services';
import type { Props } from '../Services';

describe('Services Component Snapshot', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      services: [],
    };
  });

  it('should renders correctly', () => {
    const tree = renderer
      .create(<Services {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a list of services withouth products', () => {
    props.services = [{
      name: 'Haircut',
      products: [],
    }];
    const tree = renderer
      .create(<Services {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a list of servises with products', () => {
    props.services = [{
      name: 'Haircut',
      products: [{
        productName: 'Short haircut',
        range: false,
        minDurationMinutes: '20',
        maxDurationMinutes: '60',
      }, {
        productName: 'Long haircut',
        range: false,
        minDurationMinutes: '40',
        maxDurationMinutes: '80',
      }],
    }];
    const tree = renderer
      .create(<Services {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

