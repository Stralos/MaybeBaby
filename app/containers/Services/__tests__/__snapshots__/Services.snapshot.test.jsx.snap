// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Component Snapshot should render a list of services withouth products 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.75rem;
  margin-bottom: 24px;
  font-family: 'Marcellus SC';
}

.c4 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c1 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c3 {
  margin-top: 40px;
  margin-bottom: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.4em;
  font-family: "Marcellus SC",serif;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <h3
    className="c0"
  >
    Services & Prices
  </h3>
  <ul
    className="c1"
  >
    <div
      className="c2"
    >
      <h4
        className="c3"
      >
        Haircut
      </h4>
      <ul
        className="c4"
      />
    </div>
  </ul>
</div>
`;

exports[`Services Component Snapshot should render a list of servises with products 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.75rem;
  margin-bottom: 24px;
  font-family: 'Marcellus SC';
}

.c7 {
  background: transparent;
  color: #96d7de;
  font-family: 'Marcellus SC';
  border: none;
  cursor: pointer;
  padding: 4px 8px;
  font-size: 1rem;
}

.c4 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: spaced-out;
  -webkit-justify-content: spaced-out;
  -ms-flex-pack: spaced-out;
  justify-content: spaced-out;
  border-bottom: 1px solid #555;
  box-sizing: border-box;
  padding: 8px 0;
}

.c1 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c3 {
  margin-top: 40px;
  margin-bottom: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.4em;
  font-family: "Marcellus SC",serif;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

<div>
  <h3
    className="c0"
  >
    Services & Prices
  </h3>
  <ul
    className="c1"
  >
    <div
      className="c2"
    >
      <h4
        className="c3"
      >
        Haircut
      </h4>
      <ul
        className="c4"
      >
        <li
          className="c5"
        >
          <div
            className="c6"
          >
            Short haircut
          </div>
          <button
            className="c7"
          >
            Book
          </button>
        </li>
        <li
          className="c5"
        >
          <div
            className="c6"
          >
            Long haircut
          </div>
          <button
            className="c7"
          >
            Book
          </button>
        </li>
      </ul>
    </div>
  </ul>
</div>
`;

exports[`Services Component Snapshot should renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.75rem;
  margin-bottom: 24px;
  font-family: 'Marcellus SC';
}

.c1 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

<div>
  <h3
    className="c0"
  >
    Services & Prices
  </h3>
  <ul
    className="c1"
  />
</div>
`;
